src/librustc_mir/borrow_check/mod.rs

```diff
@@ -266,6 +267,11 @@ fn do_mir_borrowck<'a, 'gcx, 'tcx>(
         if let ClearCrossCrate::Set(ref vsi) = mbcx.mir.visibility_scope_info {
             let local_decl = &mbcx.mir.local_decls[local];
 
+            // Skip closures in fn arg positions
```

Comment:
Nit: can you rephrase as "skip implicit `self` argument for closures"? I'm not sure I would understand the existing comment =)